geonode/thumbs/thumbnails.py

```diff
@@ -107,6 +107,18 @@ def create_thumbnail(
     target_crs = forced_crs.upper() if forced_crs is not None else "EPSG:3857"
 
     compute_bbox_from_datasets = False
+    is_map_with_datasets = True
+
+    if isinstance(instance, Map):
+        viable_datasets = []
+        for map_dataset in instance.datasets:
+            if map_dataset.visibility and map_dataset.local and map_dataset.ows_url:
+                # Skip no visible layer and incorrectly defined remote layer encountered (no OWS URL defined)
+                viable_datasets.append(map_dataset)
```

Comment:
So, if `viable_datasets` won't be used for other purposes, it does not make really sense to do the whole loop, store the layers into a list just to set a `boolean` later on.

Let's please optimize this code.

Reply:
makes sense. updated